servers/cu/src/domain/client/ao-process.test.js

```diff
@@ -1396,5 +1396,107 @@ describe('ao-process', () => {
     })
   })
 
-  describe.todo('saveCheckpointWith')
+  describe('saveCheckpointWith', () => {
+    const logger = createLogger('saveCheckpointWith')
+    const depsAll = {
+      logger
+    }
+
+    test('should not save checkpoint if checkpointing is disabled', async () => {
+      const saveCheckpoint = saveCheckpointWith({
+        ...depsAll,
+        address: () => {
+          assert.fail('saving checkpoint process should not have been initiated')
+        },
+        DISABLE_PROCESS_CHECKPOINT_CREATION: true
+      })
+      await saveCheckpoint({})
```

Comment:
This should have an assertion on the returned value being `undefined` or whatever is returned.